core/web/jobs_controller_test.go

```diff
@@ -92,8 +95,72 @@ func TestJobsController_Create_ValidationFailure_OffchainReportingSpec(t *testin
+	defer cleanup()
+	require.Equal(t, http.StatusOK, response.StatusCode)
+
+	jb := job.Job{}
+	require.NoError(t, app.GetDB().Preload("OffchainreportingOracleSpec").First(&jb, "type = ?", job.OffchainReporting).Error)
+
+	resource := presenters.JobResource{}
+	err = web.ParseJSONAPIResponse(cltest.ParseResponseBody(t, response), &resource)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "web oracle spec", jb.Name.ValueOrZero())
+	assert.Equal(t, jb.OffchainreportingOracleSpec.P2PPeerID, resource.OffChainReportingSpec.P2PPeerID)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.P2PBootstrapPeers, resource.OffChainReportingSpec.P2PBootstrapPeers)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.IsBootstrapPeer, resource.OffChainReportingSpec.IsBootstrapPeer)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.EncryptedOCRKeyBundleID, resource.OffChainReportingSpec.EncryptedOCRKeyBundleID)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.TransmitterAddress, resource.OffChainReportingSpec.TransmitterAddress)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.ObservationTimeout, resource.OffChainReportingSpec.ObservationTimeout)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.BlockchainTimeout, resource.OffChainReportingSpec.BlockchainTimeout)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.ContractConfigTrackerSubscribeInterval, resource.OffChainReportingSpec.ContractConfigTrackerSubscribeInterval)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.ContractConfigTrackerSubscribeInterval, resource.OffChainReportingSpec.ContractConfigTrackerSubscribeInterval)
+	assert.Equal(t, jb.OffchainreportingOracleSpec.ContractConfigConfirmations, resource.OffChainReportingSpec.ContractConfigConfirmations)
+	assert.NotNil(t, resource.PipelineSpec.DotDAGSource)
+	// Sanity check to make sure it inserted correctly
+	require.Equal(t, ethkey.EIP55Address("0x613a38AC1659769640aaE063C651F48E0250454C"), jb.OffchainreportingOracleSpec.ContractAddress)
+}
+
+func TestJobController_Create_HappyPath(t *testing.T) {
+	ethClient, sub, assertMocksCalled := cltest.NewEthMocksWithDefaultChain(t)
+	defer assertMocksCalled()
+	sub.On("Err").Return(nil)
+	sub.On("Unsubscribe").Maybe().Return()
+	h40 := eth.Head{Hash: utils.NewHash(), Number: 40}
+	h41 := eth.Head{Hash: utils.NewHash(), ParentHash: h40.Hash, Number: 41}
+	h42 := eth.Head{Hash: utils.NewHash(), ParentHash: h41.Hash, Number: 42}
+	ethClient.On("HeadByNumber", mock.Anything, mock.Anything).Return(&h42, nil)
+	ethClient.On("ChainID").Return(&cltest.FixtureChainID)
+	ethClient.On("Dial", mock.Anything).Return(nil)
+	ethClient.On("BatchCallContext", mock.Anything, mock.Anything).Return(nil)
+	ethClient.On("SubscribeNewHead", mock.Anything, mock.Anything).Return(sub, nil)
+	ethClient.On("CallContract", mock.Anything, mock.Anything, mock.Anything).Return(nil, nil)
+	ethClient.On("Close").Maybe().Return()
+	ethClient.On("CodeAt", mock.Anything, mock.Anything, mock.Anything).Return(nil, nil)
```

Comment:
Had to add this in because the job controller calls `AddJobV2`, which ends up calling `NewDelegate`. For VRF this call requires an eth client that _doesn't_ return a null head like the `NullClient` does, since it will cause a null pointer exception.

Perhaps we can handle this in the VRF code by checking `latestHead == nil` and falling back to a "reasonable" value but it's not really clear what that reasonable value should be, since this state is probably an error state in a production setting (i.e we're unable to fetch latest head of the chain we are supposedly connected to). 

I'm open to suggestions on a better way to structure this code.